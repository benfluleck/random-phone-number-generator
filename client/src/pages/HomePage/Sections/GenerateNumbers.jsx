import React from 'react'
import styled from 'styled-components';

import setClassNames from '../../../utils/setClassNames'

const GenerateNumbers = ({ title, numbers, handleOnChange, isDisabled, errorMessages }) => (
  <GenerateNumbers.Container className="section">
    <label className={
      setClassNames({
        'is-size-5': true,
        'has-text-centered': true,
        'has-text-weight-light': true,
        'label': true
      })}>
      {title} &nbsp;
    </label>
    <input
      className={
        setClassNames({
          "input": true,
          "is-medium": true,
          "is-danger": errorMessages
        })}
      type="text"
      id="addInput"
      value={numbers}
      placeholder="Number of Phone Numbers"
      onChange={handleOnChange}
      disabled={isDisabled && 'disabled'} />
    <p className="is-size-6 has-text-danger has-text-centered has-text-weight-light label">{errorMessages}</p>
  </GenerateNumbers.Container>)


GenerateNumbers.Container = styled.section`
`;


export default GenerateNumbers
