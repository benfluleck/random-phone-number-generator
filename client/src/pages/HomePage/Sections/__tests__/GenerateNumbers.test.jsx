import React from 'react';
import { shallow } from 'enzyme';

import GenerateNumbers from '../GenerateNumbers';


const setup = (props) => {
  const defaultProps = {
    title: 'Test Number',
    numbers: '35',
    handleOnChange: () => {},
    errorMessages: 'This is an error message'
  };
  const wrapper = shallow(
    <GenerateNumbers {...props} {...defaultProps} />
  );

  return {
    wrapper,
  };
};


describe('Component - GenerateNumbers', () => {
  test('should render', () => {
    const { wrapper } = setup();

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
