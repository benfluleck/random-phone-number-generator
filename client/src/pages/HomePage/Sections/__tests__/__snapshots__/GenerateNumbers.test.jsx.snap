// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - GenerateNumbers should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GenerateNumbers
    errorMessages="This is an error message"
    handleOnChange={[Function]}
    isDisabled={false}
    numbers="35"
    title="Test Number"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <label
          className="is-size-4 has-text-centered has-text-weight-light label"
        >
          Test Number
           
        </label>,
        <input
          className="input is-medium is-danger"
          disabled={false}
          id="addInput"
          onChange={[Function]}
          placeholder="Number of Phone Numbers"
          type="text"
          value="35"
        />,
        <p
          className="is-size-5 has-text-danger has-text-centered has-text-weight-light label"
        >
          This is an error message
        </p>,
      ],
      "className": "section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Test Number",
            " ",
          ],
          "className": "is-size-4 has-text-centered has-text-weight-light label",
        },
        "ref": null,
        "rendered": Array [
          "Test Number",
          " ",
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input is-medium is-danger",
          "disabled": false,
          "id": "addInput",
          "onChange": [Function],
          "placeholder": "Number of Phone Numbers",
          "type": "text",
          "value": "35",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is an error message",
          "className": "is-size-5 has-text-danger has-text-centered has-text-weight-light label",
        },
        "ref": null,
        "rendered": "This is an error message",
        "type": "p",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
",
        ],
      },
      "displayName": "styled.section",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "section",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <label
            className="is-size-4 has-text-centered has-text-weight-light label"
          >
            Test Number
             
          </label>,
          <input
            className="input is-medium is-danger"
            disabled={false}
            id="addInput"
            onChange={[Function]}
            placeholder="Number of Phone Numbers"
            type="text"
            value="35"
          />,
          <p
            className="is-size-5 has-text-danger has-text-centered has-text-weight-light label"
          >
            This is an error message
          </p>,
        ],
        "className": "section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Test Number",
              " ",
            ],
            "className": "is-size-4 has-text-centered has-text-weight-light label",
          },
          "ref": null,
          "rendered": Array [
            "Test Number",
            " ",
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input is-medium is-danger",
            "disabled": false,
            "id": "addInput",
            "onChange": [Function],
            "placeholder": "Number of Phone Numbers",
            "type": "text",
            "value": "35",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is an error message",
            "className": "is-size-5 has-text-danger has-text-centered has-text-weight-light label",
          },
          "ref": null,
          "rendered": "This is an error message",
          "type": "p",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
",
          ],
        },
        "displayName": "styled.section",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "section",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
